src/wallet/rpcwallet.cpp

```diff
@@ -3135,8 +3135,9 @@ UniValue z_getaddressforaccount(const UniValue& params, bool fHelp)
         }
     }
     if (receivers.empty()) {
+        LOCK(cs_main);
```

Comment:
@daira suspects that this likely deadlocks because in most other places in the code we lock `cs_main` before `cs_wallet`. We should just lock `cs_main` for the whole RPC method. *sigh*